vendor/github.com/keybase/gomounts/linux.go

```diff
@@ -0,0 +1,45 @@
+// +build linux
+
+package gomounts
+
+/*
+#include <stdio.h>
+#include <stdlib.h>
+#include <mntent.h>
+*/
+import "C"
+import (
+	"errors"
+	"strings"
+	"unsafe"
+)
+
+func getMountedVolumes() ([]Volume, error) {
+	result := make([]Volume, 0)
+
+	cpath := C.CString("/proc/mounts")
+	defer C.free(unsafe.Pointer(cpath))
+	cmode := C.CString("r")
+	defer C.free(unsafe.Pointer(cmode))
+	var file *C.FILE = C.setmntent(cpath, cmode)
+	if file == nil {
+		return nil, errors.New("Unable to open /proc/mounts")
+	}
+	defer C.endmntent(file)
+	var ent *C.struct_mntent
+
+	mntopt := C.CString("user_id")
+	for ent = C.getmntent(file); ent != nil; ent = C.getmntent(file) {
```

Comment:
would prefer `getmntent_r` here, too

Reply:
I've been struggling with converting this code to `getmntent_r` for a couple hours now, and have failed.  Basically it is very not obvious how to pass a `struct mntent *` to an stack-allocated `C.struct_mntent` into C-land, without rewriting the entire function is C.

Instead, I'm going to give up and just protect all calls to `GetMountedVolumes()` with a global lock in `main.go`.